README.md

```diff
@@ -0,0 +1,85 @@
+# *Ayanda*
+
+**Ayanda**  Ayanda is an Open Source Android Library that makes it easy to discover nearby devices and share files
+through a simple API. Ayanda is meant to detect nearby devices using WiFi and Bluetooth technolgy.
+
+## Usage
+ The example app in shows how to use Ayanda to discover devices on the local network, through Wifi Direct
+and through Bluetooth classic. Actions to be taken when nearby devices are discovered by each discovery method
+are defined by user defined interfaces.
+
+```java
+/* Ex: Discovering Devices on Local Network */
+
+// Define how to respond when nearby devices are discovered on network
+ILan iLan = new ILan() {
+    @Override
+    public void deviceListChanged() {
+        // Clear the list of discovered peers
+        peers.clear();
+        peerNames.clear();
+        peersAdapter.clear();
+        // update list with all peers within range
+        peers.addAll(a.lanGetDeviceList());
+        for (int i = 0; i < peers.size(); i++) {
+            Lan.Device d = (Lan.Device) peers.get(i);
+            peersAdapter.add(d.getName());
+        }
+    }
+};
+
+a = new Ayanda(this, null, iLan, null);
```

Comment:
Rather than have protocol-specific callbacks, hide all this behind a single interface. The user of the library shouldn't need to care where the service came from.

Create an interface like `AyandaProvider` and then rename your protocol-specific classes to be `BluetoothAyandaProvider`, `WifiDirectAyandaProvider`, etc. Change the `Ayanda` constructor to take in a list of providers and a single callback.